### PR TITLE
Refactor creation and deletion of GitHub hooking

### PR DESCRIPTION
**Pull request by timgluz**
Hi, 

i added a new class `GithubWebhook` to manage github hooks - it has at the moment only methods to create a hook for project and then remove it. 

I am not sure, that i'm on right way, so i stopped pimping it further.

### CODE REVIEW

**Comment by reiz** [comment]
@timgluz The overall direction is right. Just a couple tweaks are needed.

File: lib/versioneye/services/project_service.rb [hunk 1]
```diff
@@ -331,21 +331,29 @@ def self.destroy_by user, project_id
...
   end
 
 
-  def self.destroy project
+  def self.destroy project, github_token = nil
     return false if project.nil?
 
     project.children.each do |child_project|
+      # remove any attached Github hook associated with the child project
+      GithubWebhook.delete_project_hook(child_project.id, github_token)
```
Comment (timgluz) [inline lib/versioneye/services/project_service.rb]: No exceptions, it's always return boolean; exception are catched in request-functions;

Comment (reiz) [inline lib/versioneye/services/project_service.rb]: OK. Let me know when it's ready to merge.

**Comment by timgluz** [comment]
ok, ready to review

**Pull request merged**